rkt/api_service.go

```diff
@@ -114,16 +114,9 @@ func isPartOf(keyword, s string) bool {
 	return strings.Contains(s, keyword)
 }
 
-// satisfiesPodFilter returns true if the pod satisfies the filter or the filter
-// is nil, which means the pod will be returned.
-// It returns false if the pod doesn't satisfy the filter, which means it will
-// not be returned.
+// satisfiesPodFilter returns true if the pod satisfies the filter.
+// The pod, manifest, filter must not be nil.
```

Comment:
Non-pointer types, I should have said.
I understand you're not generating the pointers initially, but you're controlling what gets passed into this function. In the spirit of defensive/functional programming, IMHO it's preferable to avoid nil wherever possible; if you eliminate pointers from a function signature, then at least you've eliminated a class of problems to think about when you're writing/rewriting/debugging/testing that function, even if you haven't necessarily removed nil everywhere else in the code.
Not trying to block you here but it's something to keep in mind in general.